# Training the YOLO Classify Model
This code trains the yolo model on the frames extracted from the data set. The images must be placed into the test and train subfolders in the subfolder that corresponds with the emotion label.

In [ ]:
from ultralytics import YOLO

# Load the YOLO-pose nano model, pose version
model = YOLO('yolov8m-cls')

# Train the model with our data set defined in dogs-pose.yaml
results = model.train(data='dataset/', epochs=50, imgsz=120)

New https://pypi.org/project/ultralytics/8.1.47 available 😃 Update with 'pip install -U ultralytics'
Ultralytics YOLOv8.0.221 🚀 Python-3.10.11 torch-2.1.1+cu118 CPU (Intel Core(TM) i7-3840QM 2.80GHz)
engine\trainer: task=classify, mode=train, model=yolov8m-cls.pt, data=dataset/, epochs=50, patience=50, batch=16, imgsz=120, save=True, save_period=-1, cache=False, device=None, workers=8, project=None, name=train3, exist_ok=False, pretrained=True, optimizer=auto, verbose=True, seed=0, deterministic=True, single_cls=False, rect=False, cos_lr=False, close_mosaic=10, resume=False, amp=True, fraction=1.0, profile=False, freeze=None, overlap_mask=True, mask_ratio=4, dropout=0.0, val=True, split=val, save_json=False, save_hybrid=False, conf=None, iou=0.7, max_det=300, half=False, dnn=False, plots=True, source=None, vid_stride=1, stream_buffer=False, visualize=False, augment=False, agnostic_nms=False, classes=None, retina_masks=False, show=False, save_frames=False, save_txt=False, save_conf=Fals

train: Scanning C:\Users\Brendan\Desktop\CMPT 419 Project\YOLO model\dataset\train... 1365 images, 0 corrupt: 100%|██████████| 1365/1365 [00:00<?, ?it/s]
val: Scanning C:\Users\Brendan\Desktop\CMPT 419 Project\YOLO model\dataset\test... 176 images, 0 corrupt: 100%|██████████| 176/176 [00:00<?, ?it/s]

optimizer: 'optimizer=auto' found, ignoring 'lr0=0.01' and 'momentum=0.937' and determining best 'optimizer', 'lr0' and 'momentum' automatically... 
optimizer: AdamW(lr=0.000714, momentum=0.9) with parameter groups 38 weight(decay=0.0), 39 weight(decay=0.0005), 39 bias(decay=0.0)
Image sizes 128 train, 128 val
Using 0 dataloader workers
Logging results to runs\classify\train3
Starting training for 50 epochs...

      Epoch    GPU_mem       loss  Instances       Size



               classes   top1_acc   top5_acc: 100%|██████████| 6/6 [00:07<00:00,  1.18s/it]

                   all      0.415          1



      Epoch    GPU_mem       loss  Instances       Size


               classes   top1_acc   top5_acc: 100%|██████████| 6/6 [00:07<00:00,  1.20s/it]

                   all      0.426          1



      Epoch    GPU_mem       loss  Instances       Size


               classes   top1_acc   top5_acc: 100%|██████████| 6/6 [00:07<00:00,  1.19s/it]

                   all      0.341          1



      Epoch    GPU_mem       loss  Instances       Size


               classes   top1_acc   top5_acc: 100%|██████████| 6/6 [00:07<00:00,  1.19s/it]

                   all      0.341          1



      Epoch    GPU_mem       loss  Instances       Size


               classes   top1_acc   top5_acc: 100%|██████████| 6/6 [00:07<00:00,  1.18s/it]

                   all      0.301          1



      Epoch    GPU_mem       loss  Instances       Size


               classes   top1_acc   top5_acc: 100%|██████████| 6/6 [00:07<00:00,  1.21s/it]

                   all      0.398          1



      Epoch    GPU_mem       loss  Instances       Size


               classes   top1_acc   top5_acc: 100%|██████████| 6/6 [00:07<00:00,  1.20s/it]

                   all      0.324          1



      Epoch    GPU_mem       loss  Instances       Size


               classes   top1_acc   top5_acc: 100%|██████████| 6/6 [00:07<00:00,  1.19s/it]

                   all      0.352          1



      Epoch    GPU_mem       loss  Instances       Size


               classes   top1_acc   top5_acc: 100%|██████████| 6/6 [00:07<00:00,  1.19s/it]

                   all      0.358          1



      Epoch    GPU_mem       loss  Instances       Size


      10/50         0G    0.02332          5        128: 100%|██████████| 86/86 [03:37<00:00,  2.53s/it]
               classes   top1_acc   top5_acc: 100%|██████████| 6/6 [00:07<00:00,  1.17s/it]

                   all      0.369          1



      Epoch    GPU_mem       loss  Instances       Size


      11/50         0G   0.009344          5        128: 100%|██████████| 86/86 [03:35<00:00,  2.50s/it]
               classes   top1_acc   top5_acc: 100%|██████████| 6/6 [00:07<00:00,  1.17s/it]

                   all      0.438          1



      Epoch    GPU_mem       loss  Instances       Size


      12/50         0G   0.003157          5        128: 100%|██████████| 86/86 [03:35<00:00,  2.50s/it]
               classes   top1_acc   top5_acc: 100%|██████████| 6/6 [00:07<00:00,  1.22s/it]

                   all      0.369          1



      Epoch    GPU_mem       loss  Instances       Size


      13/50         0G   0.002263          5        128: 100%|██████████| 86/86 [03:36<00:00,  2.51s/it]
               classes   top1_acc   top5_acc: 100%|██████████| 6/6 [00:07<00:00,  1.20s/it]

                   all      0.375          1



      Epoch    GPU_mem       loss  Instances       Size


      14/50         0G   0.002403          5        128: 100%|██████████| 86/86 [03:35<00:00,  2.50s/it]
               classes   top1_acc   top5_acc: 100%|██████████| 6/6 [00:07<00:00,  1.19s/it]

                   all      0.369          1



      Epoch    GPU_mem       loss  Instances       Size


      15/50         0G   0.002764          5        128: 100%|██████████| 86/86 [03:48<00:00,  2.65s/it]
               classes   top1_acc   top5_acc: 100%|██████████| 6/6 [00:07<00:00,  1.24s/it]

                   all      0.301          1



      Epoch    GPU_mem       loss  Instances       Size


      16/50         0G   0.003085          5        128: 100%|██████████| 86/86 [04:04<00:00,  2.84s/it]
               classes   top1_acc   top5_acc: 100%|██████████| 6/6 [00:07<00:00,  1.33s/it]

                   all      0.443          1



      Epoch    GPU_mem       loss  Instances       Size


      17/50         0G    0.02174          5        128: 100%|██████████| 86/86 [03:50<00:00,  2.68s/it]
               classes   top1_acc   top5_acc: 100%|██████████| 6/6 [00:07<00:00,  1.28s/it]

                   all      0.415          1



      Epoch    GPU_mem       loss  Instances       Size


      18/50         0G    0.01819          5        128: 100%|██████████| 86/86 [03:54<00:00,  2.73s/it]
               classes   top1_acc   top5_acc: 100%|██████████| 6/6 [00:07<00:00,  1.32s/it]

                   all      0.409          1



      Epoch    GPU_mem       loss  Instances       Size


      19/50         0G    0.01656          5        128: 100%|██████████| 86/86 [03:54<00:00,  2.73s/it]
               classes   top1_acc   top5_acc: 100%|██████████| 6/6 [00:07<00:00,  1.26s/it]

                   all      0.438          1



      Epoch    GPU_mem       loss  Instances       Size


      20/50         0G    0.02108          5        128: 100%|██████████| 86/86 [03:52<00:00,  2.70s/it]
               classes   top1_acc   top5_acc: 100%|██████████| 6/6 [00:07<00:00,  1.21s/it]

                   all      0.381          1



      Epoch    GPU_mem       loss  Instances       Size


      21/50         0G    0.01305          5        128: 100%|██████████| 86/86 [03:37<00:00,  2.52s/it]
               classes   top1_acc   top5_acc: 100%|██████████| 6/6 [00:07<00:00,  1.22s/it]

                   all       0.42          1



      Epoch    GPU_mem       loss  Instances       Size


      22/50         0G    0.01726          5        128: 100%|██████████| 86/86 [03:37<00:00,  2.52s/it]
               classes   top1_acc   top5_acc: 100%|██████████| 6/6 [00:07<00:00,  1.19s/it]

                   all      0.432          1



      Epoch    GPU_mem       loss  Instances       Size


      23/50         0G    0.01327          5        128: 100%|██████████| 86/86 [03:35<00:00,  2.51s/it]
               classes   top1_acc   top5_acc: 100%|██████████| 6/6 [00:07<00:00,  1.19s/it]

                   all      0.369          1



      Epoch    GPU_mem       loss  Instances       Size


      24/50         0G    0.01156          5        128: 100%|██████████| 86/86 [03:56<00:00,  2.76s/it]
               classes   top1_acc   top5_acc: 100%|██████████| 6/6 [00:07<00:00,  1.22s/it]

                   all      0.375          1



      Epoch    GPU_mem       loss  Instances       Size


      25/50         0G   0.003591          5        128: 100%|██████████| 86/86 [03:59<00:00,  2.79s/it]
               classes   top1_acc   top5_acc: 100%|██████████| 6/6 [00:07<00:00,  1.28s/it]

                   all       0.33          1



      Epoch    GPU_mem       loss  Instances       Size


      26/50         0G   0.003562          5        128: 100%|██████████| 86/86 [03:49<00:00,  2.67s/it]
               classes   top1_acc   top5_acc: 100%|██████████| 6/6 [00:07<00:00,  1.30s/it]

                   all      0.409          1



      Epoch    GPU_mem       loss  Instances       Size


      27/50         0G   0.005908          5        128: 100%|██████████| 86/86 [03:55<00:00,  2.73s/it]
               classes   top1_acc   top5_acc: 100%|██████████| 6/6 [00:07<00:00,  1.28s/it]

                   all      0.426          1



      Epoch    GPU_mem       loss  Instances       Size


      28/50         0G   0.003252          5        128: 100%|██████████| 86/86 [03:57<00:00,  2.76s/it]
               classes   top1_acc   top5_acc: 100%|██████████| 6/6 [00:07<00:00,  1.26s/it]

                   all      0.438          1



      Epoch    GPU_mem       loss  Instances       Size


      29/50         0G   0.002348          5        128: 100%|██████████| 86/86 [03:51<00:00,  2.69s/it]
               classes   top1_acc   top5_acc: 100%|██████████| 6/6 [00:07<00:00,  1.29s/it]

                   all      0.432          1



      Epoch    GPU_mem       loss  Instances       Size


      30/50         0G  0.0009074          5        128: 100%|██████████| 86/86 [03:48<00:00,  2.66s/it]
               classes   top1_acc   top5_acc: 100%|██████████| 6/6 [00:07<00:00,  1.30s/it]

                   all       0.46          1



      Epoch    GPU_mem       loss  Instances       Size


      31/50         0G   0.001067          5        128: 100%|██████████| 86/86 [03:59<00:00,  2.79s/it]
               classes   top1_acc   top5_acc: 100%|██████████| 6/6 [00:07<00:00,  1.32s/it]

                   all       0.46          1



      Epoch    GPU_mem       loss  Instances       Size


      32/50         0G    0.00699          5        128: 100%|██████████| 86/86 [03:54<00:00,  2.72s/it]
               classes   top1_acc   top5_acc: 100%|██████████| 6/6 [00:08<00:00,  1.36s/it]

                   all      0.489          1



      Epoch    GPU_mem       loss  Instances       Size


      33/50         0G   0.001235          5        128: 100%|██████████| 86/86 [03:47<00:00,  2.65s/it]
               classes   top1_acc   top5_acc: 100%|██████████| 6/6 [00:07<00:00,  1.24s/it]

                   all       0.46          1



      Epoch    GPU_mem       loss  Instances       Size


      34/50         0G  0.0004397          5        128: 100%|██████████| 86/86 [03:44<00:00,  2.61s/it]
               classes   top1_acc   top5_acc: 100%|██████████| 6/6 [00:07<00:00,  1.23s/it]

                   all      0.494          1



      Epoch    GPU_mem       loss  Instances       Size


      35/50         0G  0.0008674          5        128: 100%|██████████| 86/86 [03:42<00:00,  2.59s/it]
               classes   top1_acc   top5_acc: 100%|██████████| 6/6 [00:07<00:00,  1.24s/it]

                   all      0.466          1



      Epoch    GPU_mem       loss  Instances       Size


      36/50         0G  0.0003795          5        128: 100%|██████████| 86/86 [03:42<00:00,  2.59s/it]
               classes   top1_acc   top5_acc: 100%|██████████| 6/6 [00:07<00:00,  1.24s/it]

                   all      0.438          1



      Epoch    GPU_mem       loss  Instances       Size


      37/50         0G  0.0003526          5        128: 100%|██████████| 86/86 [03:46<00:00,  2.63s/it]
               classes   top1_acc   top5_acc: 100%|██████████| 6/6 [00:07<00:00,  1.29s/it]

                   all      0.472          1



      Epoch    GPU_mem       loss  Instances       Size


      38/50         0G   0.001033          5        128: 100%|██████████| 86/86 [03:46<00:00,  2.63s/it]
               classes   top1_acc   top5_acc: 100%|██████████| 6/6 [00:07<00:00,  1.20s/it]

                   all      0.443          1



      Epoch    GPU_mem       loss  Instances       Size


      39/50         0G    0.00656          5        128: 100%|██████████| 86/86 [03:46<00:00,  2.63s/it]
               classes   top1_acc   top5_acc: 100%|██████████| 6/6 [00:07<00:00,  1.21s/it]

                   all       0.42          1



      Epoch    GPU_mem       loss  Instances       Size


      40/50         0G  0.0004614          5        128: 100%|██████████| 86/86 [04:24<00:00,  3.08s/it]
               classes   top1_acc   top5_acc: 100%|██████████| 6/6 [00:07<00:00,  1.32s/it]

                   all      0.432          1



      Epoch    GPU_mem       loss  Instances       Size


      41/50         0G  0.0005973          5        128: 100%|██████████| 86/86 [03:43<00:00,  2.60s/it]
               classes   top1_acc   top5_acc: 100%|██████████| 6/6 [00:07<00:00,  1.21s/it]

                   all      0.415          1



      Epoch    GPU_mem       loss  Instances       Size


      42/50         0G  0.0009886          5        128: 100%|██████████| 86/86 [03:36<00:00,  2.51s/it]
               classes   top1_acc   top5_acc: 100%|██████████| 6/6 [00:07<00:00,  1.20s/it]

                   all      0.392          1



      Epoch    GPU_mem       loss  Instances       Size


      43/50         0G   0.000376          5        128: 100%|██████████| 86/86 [03:34<00:00,  2.50s/it]
               classes   top1_acc   top5_acc: 100%|██████████| 6/6 [00:07<00:00,  1.21s/it]

                   all      0.398          1



      Epoch    GPU_mem       loss  Instances       Size


      44/50         0G  0.0003894          5        128: 100%|██████████| 86/86 [03:36<00:00,  2.51s/it]
               classes   top1_acc   top5_acc: 100%|██████████| 6/6 [00:07<00:00,  1.20s/it]

                   all      0.375          1



      Epoch    GPU_mem       loss  Instances       Size


      45/50         0G  0.0003623          5        128: 100%|██████████| 86/86 [03:35<00:00,  2.51s/it]
               classes   top1_acc   top5_acc: 100%|██████████| 6/6 [00:06<00:00,  1.16s/it]

                   all       0.42          1



      Epoch    GPU_mem       loss  Instances       Size


      46/50         0G  0.0001786          5        128: 100%|██████████| 86/86 [03:35<00:00,  2.50s/it]
               classes   top1_acc   top5_acc: 100%|██████████| 6/6 [00:07<00:00,  1.18s/it]

                   all      0.449          1



      Epoch    GPU_mem       loss  Instances       Size


      47/50         0G  0.0009382          5        128: 100%|██████████| 86/86 [03:41<00:00,  2.58s/it]
               classes   top1_acc   top5_acc: 100%|██████████| 6/6 [00:07<00:00,  1.17s/it]

                   all      0.403          1



      Epoch    GPU_mem       loss  Instances       Size


      48/50         0G  0.0002476          5        128: 100%|██████████| 86/86 [03:36<00:00,  2.52s/it]
               classes   top1_acc   top5_acc: 100%|██████████| 6/6 [00:07<00:00,  1.19s/it]

                   all      0.392          1



      Epoch    GPU_mem       loss  Instances       Size


      49/50         0G  0.0004022          5        128: 100%|██████████| 86/86 [03:34<00:00,  2.50s/it]
               classes   top1_acc   top5_acc: 100%|██████████| 6/6 [00:07<00:00,  1.17s/it]

                   all      0.426          1



      Epoch    GPU_mem       loss  Instances       Size


      50/50         0G   0.000379          5        128: 100%|██████████| 86/86 [03:34<00:00,  2.49s/it]
               classes   top1_acc   top5_acc: 100%|██████████| 6/6 [00:07<00:00,  1.18s/it]

                   all      0.381          1



50 epochs completed in 3.228 hours.
Optimizer stripped from runs\classify\train3\weights\last.pt, 31.7MB
Optimizer stripped from runs\classify\train3\weights\best.pt, 31.7MB

Validating runs\classify\train3\weights\best.pt...
Ultralytics YOLOv8.0.221 🚀 Python-3.10.11 torch-2.1.1+cu118 CPU (Intel Core(TM) i7-3840QM 2.80GHz)
YOLOv8m-cls summary (fused): 103 layers, 15767780 parameters, 0 gradients, 41.6 GFLOPs
WARNING ⚠️ Dataset 'split=val' not found, using 'split=test' instead.
train: C:\Users\Brendan\Desktop\CMPT 419 Project\YOLO model\dataset\train... found 1365 images in 4 classes ✅ 
val: None...
test: C:\Users\Brendan\Desktop\CMPT 419 Project\YOLO model\dataset\test... found 176 images in 4 classes ✅ 


               classes   top1_acc   top5_acc: 100%|██████████| 6/6 [00:07<00:00,  1.32s/it]


                   all      0.489          1
Speed: 0.0ms preprocess, 43.5ms inference, 0.0ms loss, 0.0ms postprocess per image
Results saved to runs\classify\train3
Results saved to runs\classify\train3


### Training Summary.
The model train for 50 epochs over 3.22 hours. The accuracy is 48.9%